ion/tokenizer.go

```diff
@@ -582,20 +585,27 @@ func (t *tokenizer) readLongString() (string, error) {
 		if err != nil {
 			return "", err
 		}
+		if isInvalidChar(c) {
+			return "", &SyntaxError{"Invalid character", t.pos}
+		}
 
 		switch c {
 		case -1:
 			return "", t.invalidChar(c)
 
 		case '\'':
+			startPosition := t.pos
 			ok, err := t.skipEndOfLongString(t.skipCommentsHandler)
 			if err != nil {
 				return "", err
 			}
 			if ok {
 				return ret.String(), nil
 			}
-
+			if startPosition == t.pos {
```

Comment:
It looks like [`skipEndOfLongString` reports whether it consumed the string ending](https://github.com/amzn/ion-go/blob/64990f801f0f53d08a2d7cdb6f474c5a19fbb5ff/ion/skipper.go#L534-L537), so I think you can replace this position check with an `else {...}` attached to the `if ok {...}`.

```go
wasConsumed, err := t.skipEndOfLongString(t.skipCommentsHandler)
if err != nil {
	return "", err
}
if wasConsumed {
	return ret.String(), nil
} else {
        // The ' was not part of a long string ending.
        ret.writeByte(byte(c))
}
```

Reply:
It'll also return false if it skipped over a `'''   /* pause in the string */   '''` sequence, in which case you don't want to keep the '. :( This seems correct to me in the short term, `skipEndOfLongString` should probably should be refactored to return a tri-state in the longer term.

Reply:
Yes, I almost refactored ```skipEndOfLongString``` but then I decided to KISS.
I can create an issue for it.